pkg/trace/api/api.go

```diff
@@ -294,12 +316,29 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 			r.wg.Done()
 			watchdog.LogOnPanic()
 		}()
-		r.processTraces(ts, traces)
+		entityID := req.Header.Get(headerEntityID)
+		r.processTraces(ts, entityID, traces)
 	}()
 }
 
-func (r *HTTPReceiver) processTraces(ts *info.TagStats, traces pb.Traces) {
+// Trace specifies information about a trace received by the API.
+type Trace struct {
+	// Source specifies information about the source of these traces, such as:
+	// language, interpreter, tracer version, etc.
+	Source *info.Tags
+
+	// EntityTags specifies orchestrator tags corresponding to the origin of this
+	// trace (e.g. K8S pod, Docker image, ECS, etc).
+	EntityTags map[string]string
+
+	// Spans holds the spans of this trace.
+	Spans pb.Trace
+}
+
+func (r *HTTPReceiver) processTraces(ts *info.TagStats, entityID string, traces pb.Traces) {
 	defer timing.Since("datadog.trace_agent.internal.normalize_ms", time.Now())
+
+	entityTags := getEntityTags(entityID)
```

Comment:
Are we making this call even if the entity id is an empty value?

Reply:
Yes. If it's empty, `tagger.Tag` returns an error, in which case `entityTags` will be an empty map.

Reply:
OK so `tagger.Tag` will not make a real call but will just say "it's empty I will not do anything" right?

Reply:
https://github.com/DataDog/datadog-agent/blob/e5b2695dacf0cbe25e066cfe7fd58f30daaf7b85/pkg/tagger/tagger.go#L225-L227

Reply:
Resolving.